packages/docusaurus/src/webpack/utils.ts

```diff
@@ -284,137 +284,157 @@ export function compile(config: Configuration[]): Promise<void> {
+  const assetResourceQuery = /asset/;
+  // Can this be removed? see https://github.com/facebook/docusaurus/commit/2f21d306bdd4d286cc5d25c81adaea2fc77f0474#commitcomment-50223144)
+  const notAssetResourceQuery: RuleSetRule['resourceQuery'] = {not: [/asset/]};
 
   // defines the path/pattern of the assets handled by webpack
-  const fileLoaderFileName = (folder: AssetFolder) =>
-    `${OUTPUT_STATIC_ASSETS_DIR_NAME}/${folder}/[name]-[hash].[ext]`;
-
-  const loaders: FileLoaderUtils['loaders'] = {
-    file: (options: {folder: AssetFolder}) => {
-      return {
-        loader: require.resolve(`file-loader`),
-        options: {
-          name: fileLoaderFileName(options.folder),
-        },
-      };
-    },
-    url: (options: {folder: AssetFolder}) => {
-      return {
-        loader: require.resolve(`url-loader`),
-        options: {
-          limit: urlLoaderLimit,
-          name: fileLoaderFileName(options.folder),
-          fallback: require.resolve(`file-loader`),
+  function fileNameGenerator(folder: AssetFolder) {
+    return {
+      filename: `${OUTPUT_STATIC_ASSETS_DIR_NAME}/${folder}/[name]-[hash][ext]`,
+    };
+  }
+
+  function baseAssetRule(folder: AssetFolder): RuleSetRule {
+    return {
+      type: 'asset',
+      parser: {
+        dataUrlCondition: {
+          // Threshold for datauri/file (previously set on url-loader)
+          // files/images < 10kb will be inlined as base64 strings directly in the JS bundle
+          // See https://webpack.js.org/guides/asset-modules/#general-asset-type
+          maxSize: 10 * 1024,
         },
-      };
-    },
+      },
+      generator: fileNameGenerator(folder),
+      resourceQuery: notAssetResourceQuery,
+    };
+  }
 
-    // TODO find a better solution to avoid conflicts with the ideal-image plugin
-    // TODO this may require a little breaking change for ideal-image users?
-    // Maybe with the ideal image plugin, all md images should be "ideal"?
-    // This is used to force url-loader+file-loader on markdown images
-    // https://webpack.js.org/concepts/loaders/#inline
-    inlineMarkdownImageFileLoader: `!url-loader?limit=${urlLoaderLimit}&name=${fileLoaderFileName(
-      'images',
-    )}&fallback=file-loader!`,
-    inlineMarkdownLinkFileLoader: `!file-loader?name=${fileLoaderFileName(
-      'files',
-    )}!`,
-  };
+  function imageAssetRule(): RuleSetRule {
+    return {
+      ...baseAssetRule('images'),
+      test: /\.(ico|jpg|jpeg|png|gif|webp)(\?.*)?$/,
+    };
+  }
 
-  const rules: FileLoaderUtils['rules'] = {
-    /**
-     * Loads image assets, inlines images via a data URI if they are below
-     * the size threshold
-     */
-    images: () => {
-      return {
-        use: [loaders.url({folder: 'images'})],
-        test: /\.(ico|jpg|jpeg|png|gif|webp)(\?.*)?$/,
-      };
-    },
+  function fontAssetRule(): RuleSetRule {
+    return {
+      ...baseAssetRule('fonts'),
+      test: /\.(woff|woff2|eot|ttf|otf)$/,
```

Comment:
Small note - extensions can be `woff`, `WOFF`, so I prefer to use `i` flag for all regexp, but this not related to this PR :smile:

Reply:
thanks :)